STYLEGUIDE.md

```diff
@@ -0,0 +1,76 @@
+# Helm Chart Style Guide
+
+Due to the nature of helm, there are many ways to accomplish the same goal. There are also sometimes unintended side effects when chosing a particular style over another.  
+
+## Be explicit 
```

Comment:
I checked with @nadegepepin; our in-house style is to use Sentence Case for headings, so should probably use it throughout this document (and elsewhere, to include the `README.md`).